src/coreComponents/dataRepository/xmlWrapper.hpp

```diff
@@ -61,6 +59,179 @@ using xmlAttribute = pugi::xml_attribute;
+  size_t const offsetInLine;
+  /// Character offset of this element in the file that contains it (starting from 0)
+  /// Equals to xmlDocument::npos if it couldn't be determined.
+  size_t const offset;
+
+  /**
+   * @brief Constructor of this struct.
+   */
+  xmlAttributePos( string const & filePath, size_t line, size_t offsetInLine, size_t offset );
+  /**
+   * @return false if the position is undetermined.
+   */
+  bool isFound() const;
+  /**
+   * @return returns a string to locate the node (typically the source file and line).
+   */
+  string toString() const;
+};
+
+/// Stores the source document, file path, and position in the file of a xml node.
+/// It can help to retrieve the position (xmlAttributePos) of a xml attribute.
+struct xmlNodePos : xmlAttributePos
+{
+  /// Reference of the main xmlDocument that contains all original file buffers.
+  xmlDocument const & document;
+
+  /**
+   * @brief Constructor of this struct.
+   */
+  xmlNodePos( xmlDocument const & document, string const & filePath, size_t line, size_t offsetInLine, size_t offset );
+  /**
+   * @return false if the position is undetermined.
+   */
+  bool isFound() const;
+  /**
+   * @brief Compute the xmlAttributePos of the attributed with the given name.
+   */
+  xmlAttributePos getAttributeLine( string const & attName ) const;
+};
+
+/// Wrapper class for the type of xml document.
+/// This class exists to intercept file / string loading methods, and to keep the loaded buffers,
+/// in order to retrieve the source file and line of nodes and attributes.
+class xmlDocument : public pugi::xml_document
+{
+public:
+  /// Error value for when an offset / line position is undetermined.
+  static const size_t npos;
+
+  /**
+   * @brief Construct an empty xmlDocument that waits to load something.
+   */
+  xmlDocument();
+
+  /**
+   * @brief Get the original file buffer loaded during the last load_X() call on this object.
+   */
+  string const & getOriginalBuffer() const;
+  /**
+   * @brief If the specified file at the "filePath" is the loaded document of the instance,
+   * or one of its includes, returns its original buffer as a string.
+   * Returns nullptr if filePath is not a loaded and available document.
+   */
+  string const * getOriginalBuffer( string const & filePath ) const;
+  /**
+   * @brief Map containing the original buffers of the document and its includes, indexed by file path.
+   */
+  map< string, string > const & getOriginalBuffers() const;
+  /**
+   * @brief If load_file() has been loaded, returns the path of the source file.
+   * If another load method has been called, it returns a generated unique value.
+   */
+  string const & getFilePath() const;
+  /**
+   * @brief Compute the position of the given node if the node file information are loaded.
+   * @throws an InputError if the node position couldn't be determined despite the node source file
+   * informations being loaded.
```

Comment:
@brief Compute the position of the given node.
 @throws an InputError if the node position is not found and source file is loaded.